Three modes for weather (#1616)
home-assistant/developers.home-assistant: docs/core/entity/weather.md

### Forecast

> Forecast data can be daily, hourly or twice_daily in it's respective array. An integration can set and use any/all of them.

Reviewer: ```suggestion
Forecast data can be daily, hourly or twice_daily in its respective array. An integration can set and use any/all of them.
```

| wind_bearing | float or string | `None` | The wind bearing in azimuth angle (degrees) or 1-3 letter cardinal direction.
| native_wind_gust_speed | int | `None` | The wind gust speed in m/s, km/h, mi/h, ft/s or kn.
| native_wind_speed | int | `None` | The wind speed in m/s, km/h, mi/h, ft/s or kn.

> | is_daytime | bool | `None` | For `async_forecast_twice_daily` required to use for day/night.

Reviewer 2: ```suggestion
| is_daytime | bool | `None` | This is mandatory in forecast data returned by `async_forecast_twice_daily` to indicate day or night.
```

Forecast data needs to follow the same unit of measurement as defined for properties where applicable.

### Methods to get weather forecast(s)

These method are called to fetch forecasts from the api.

```python
class MyWeatherEntity(WeatherEntity):
    """Represent a Weather entity."""

    async def async_forecast_daily(self) -> list[Forecast] | None:
        """Return the daily forecast in native units.
        

>         Only implement this method if `FORECAST_DAILY` is set

Author: ```suggestion
        Only implement this method if `WeatherEntityFeature.FORECAST_DAILY` is set
```

        """

    async def async_forecast_twice_daily(self) -> list[Forecast] | None:
        """Return the twice daily forecast in native units.
        

>         Only implement this method if `FORECAST_TWICE_DAILY` is set

Author: ```suggestion
        Only implement this method if `WeatherEntityFeature.FORECAST_TWICE_DAILY` is set
```

        """

    async def async_forecast_hourly(self) -> list[Forecast] | None:
        """Return the hourly forecast in native units.
        

>         Only implement this method if `FORECAST_HOURLY` is set

Author: ```suggestion
        Only implement this method if `WeatherEntityFeature.FORECAST_HOURLY` is set
```
